app/v2/connectors/PersistenceConnector.scala

```diff
@@ -177,18 +177,18 @@ class PersistenceConnectorImpl @Inject() (httpClientV2: HttpClientV2, appConfig:
       }
   }
 
-  override def getMovements(eori: EORINumber, movementType: MovementType, updatedSince: Option[OffsetDateTime])(implicit
+  override def getMovements(eori: EORINumber, movementType: MovementType, updatedSince: Option[OffsetDateTime], movementEORI: Option[EORINumber])(implicit
     hc: HeaderCarrier,
     ec: ExecutionContext
   ): Future[Seq[MovementSummary]] = {
-    val url = withDateTimeParameter(
+    val urlWithDateOption = withDateTimeParameter(
```

Comment:
can we please move the logic in withDateTimeParameter and withMovementEORIParameter to the V2BaseConnector, so that we can call getAllMovementsUrl with the movementEori and updatedSince and get the path back?

Also, do we even need those separate methods or does addQueryParam consider Optional values

Reply:
I don't think so that is better way to do this to move to V2BaseConnector. As with V2BaseConnector with all the api we are getting the base path(UrlPath). Same way it is done for others like for updatedSince and receivedSince optional param we have used those optional parameters in PersistenceConnector. Also if we are moving withDateTimeParameter and withMovementEORIParameter to V2BaseConnector it will little bit complicate the things.
So I would say better to keep it as.

Yes we need two separate method as in case of we are not passing movementEORI it should call base uri without movementEORI. But when I tried with **withQueryParam** and in case of not passing any value to movementEORI it is appending movementEORI with the request url and so not returning anything from backend api.

Reply:
I see what you mean about it not being a good idea to move it to the BaseConnector, but I still think we can simplify the code by using patten matching:

`   val url = appConfig.movementsUrl.withPath(getAllMovementsUrl(eori, movementType))

    val urlWithQueryParams = (updatedSince, movementEORI) match {
      case (Some(datetime), Some(eori)) => url.addParams("updatedSince" -> DateTimeFormatter.ISO_DATE_TIME.format(datetime), "movementEORI" -> eori.value)
      case (Some(datetime), None)       => url.addParams("updatedSince" -> DateTimeFormatter.ISO_DATE_TIME.format(datetime))
      case (None, Some(eori))           => url.addParams("movementEORI" -> eori.value)
      case (None, None)                 => url
    }`

Reply:
done as suggested. Thanks